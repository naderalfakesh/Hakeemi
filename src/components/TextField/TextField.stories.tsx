import React, { useState } from 'react';
import { View, StyleSheet } from 'react-native';
import { storiesOf } from '@storybook/react-native';
import TextField from './TextField';

const Container = (storyFn: any) => (
  <View style={styles.container}>{storyFn()}</View>
);

const TextFieldDemo = () => {
  const [value, setValue] = useState('');
  return (
    <TextField
      placeholder="PlaceHolder"
      value={value}
      onChangeText={setValue}
    />
  );
};

storiesOf('Text field', module)
  .addDecorator(Container)
  .add('default', () => <TextFieldDemo />);

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    padding: 50,
  },
});
