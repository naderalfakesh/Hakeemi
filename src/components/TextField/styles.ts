import { StyleSheet } from 'react-native';
import colors from '../../theme/colors';

export default StyleSheet.create({
  base: {
    color: colors.black,
    borderBottomColor: colors.dark,
    borderBottomWidth: 1,
    fontSize: 18,
    lineHeight: 22,
    padding: 10,
  },
});
