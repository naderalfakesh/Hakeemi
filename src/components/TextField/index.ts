import TextField from './TextField';

export type { Props } from './TextField';

export default TextField;
