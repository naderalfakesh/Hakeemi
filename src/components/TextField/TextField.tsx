import React, { FC } from 'react';
import { TextInput, TextStyle, TextInputProps } from 'react-native';
import colors from '../../theme/colors';
import styles from './styles';

export interface Props extends TextInputProps {
  style?: TextStyle;
  placeholder?: string;
  value: string;
  onChangeText: (value: string) => void;
}

const TextField: FC<Props> = ({
  style,
  placeholder,
  value,
  onChangeText,
  ...otherProps
}) => {
  return (
    <TextInput
      testID="text-field"
      style={[styles.base, style]}
      value={value}
      placeholder={placeholder}
      placeholderTextColor={colors.dark}
      onChangeText={onChangeText}
      {...otherProps}
    />
  );
};

export default TextField;
