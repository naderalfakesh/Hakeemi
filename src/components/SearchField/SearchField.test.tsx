import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import TextField from './SearchField';

const mockFn = jest.fn();
describe('Button component test', () => {
  it('renders correctly with default params', () => {
    const { toJSON } = render(
      <TextField placeholder="placeHolder" value="" onChangeText={mockFn} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('invokes function on text change', () => {
    const { getByTestId } = render(
      <TextField value="Hello" onChangeText={mockFn} />,
    );
    const field = getByTestId('text-field');
    fireEvent.changeText(field, 'world');
    expect(mockFn).toBeCalledWith('world');
  });

  it('invokes function on search press', () => {
    const { getByTestId } = render(
      <TextField value="Hello" onChangeText={mockFn} handleSearch={mockFn} />,
    );
    const search = getByTestId('search-icon');
    fireEvent.press(search);
    expect(mockFn).toBeCalledWith('Hello');
  });

  it('shows clear icon when input is dirty and invokes function on clear icon press', () => {
    const { rerender, getByTestId, queryByTestId } = render(
      <TextField value="" onChangeText={mockFn} handleClear={mockFn} />,
    );

    expect(queryByTestId('clear-icon')).toBeFalsy();

    rerender(
      <TextField value="text" onChangeText={mockFn} handleClear={mockFn} />,
    );

    const clear = getByTestId('clear-icon');
    fireEvent.press(clear);
    expect(mockFn).toBeCalled();
  });
});
