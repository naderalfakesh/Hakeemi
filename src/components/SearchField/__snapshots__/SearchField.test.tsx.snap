// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test renders correctly with default params 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(237, 243, 247, 1)",
        "borderRadius": 30,
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "rgba(0, 188, 212, 1)",
          "fontSize": 20,
        },
        Array [
          Object {
            "lineHeight": 20,
            "textAlign": "center",
          },
          Object {
            "marginRight": 10,
            "padding": 2,
          },
        ],
        Object {
          "fontFamily": "doctor-font",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="search-icon"
  >
    î Œ
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[MockFunction]}
    placeholder="placeHolder"
    placeholderTextColor="rgba(144, 144, 144, 1)"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "rgba(0, 0, 0, 1)",
        "flexGrow": 1,
        "fontSize": 18,
        "lineHeight": 24,
      }
    }
    testID="text-field"
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
